operations/mimir/autoscaling.libsonnet

```diff
@@ -215,33 +217,40 @@
     )
   |||,
 
-
-  newQueryFrontendScaledObject(
+  newResourceScaledObject(
     name,
     cpu_requests,
     memory_requests,
     min_replicas,
     max_replicas,
     cpu_target_utilization,
     memory_target_utilization,
+    with_cortex_prefix=false,
+    weight=1,
+    scaledownPeriod=null,
   ):: self.newScaledObject(
     name, $._config.namespace, {
-      min_replica_count: min_replicas,
-      max_replica_count: max_replicas,
+      min_replica_count: replicasWithWeight(min_replicas, weight),
+      max_replica_count: replicasWithWeight(max_replicas, weight),
+
+      [if scaledownPeriod != null then 'scaledownPeriod']: scaledownPeriod,
+
       triggers: [
         {
-          metric_name: '%s_cpu_hpa_%s' % [std.strReplace(name, '-', '_'), $._config.namespace],
+          metric_name: '%s%s_cpu_hpa_%s' %
+                       ([if with_cortex_prefix then 'cortex_' else ''] + [std.strReplace(name, '-', '_'), $._config.namespace]),
 
-          query: cpuHPAQuery % {
+          query: metricWithWeight(cpuHPAQuery % {
             container: name,
             namespace: $._config.namespace,
-          },
+          }, weight),
 
           // Threshold is expected to be a string
           threshold: std.toString(std.floor(cpuToMilliCPUInt(cpu_requests) * cpu_target_utilization)),
         },
         {
-          metric_name: '%s_memory_hpa_%s' % [std.strReplace(name, '-', '_'), $._config.namespace],
+          metric_name: '%s%s_memory_hpa_%s' %
+                       ([if with_cortex_prefix then 'cortex_' else ''] + [std.strReplace(name, '-', '_'), $._config.namespace]),
 
           query: memoryHPAQuery % {
```

Comment:
Do we need to apply `metricWithWeight` here as well?

Reply:
There's no place where `metricWithWeight` was previously applied to a memory HPA query, so I didn't add it here to maintain historical consistency. It may be more correct to do so for "expected behavior" reasons, though!

Reply:
If that's how it's always been, then that's fine with me.